### DISCUSSION TITLE
FSR3 Frame Gen for Native FSR3.0 games

### DISCUSSION THREAD

**Discussion by david-dorner** [optiscaler/OptiScaler#67] [Ideas]
I have tried using this mod as a method of simply swapping FSR3.0 for 3.1 in games to get the better upscaling. One issue that I have faced is that I cannot enable frame gen when doing that (I have tried the 0.7 version, but the frame gen has a lot of artifacts in most games). Is there a way the mod could maybe check if Native FSR frame gen is already available and then use that for the frame gen instead? I know it's not that big of a deal but then you could use this mod as a way to simply swap FSR3.0 to FSR3.1

**Comment by cdozdil** [comment]
Currently mod only supports DLSS inputs but it's planned to add other upscalers as inputs too.

**Comment by Fr0stX76** [comment]
That would be nice, in my case I would like to be able to use FSR-FG with DLSS upscaler (Nvidia GPU) when the game only support FSR upscaler & FG 3.0. In some games I prefer FSR-FG to DLSS-FG, but I find DLSS upscaler better (my opinion).

Currently this is only possible if the game support FSR 3.1, as the FG can be decoupled from the upscaler. Not on FSR 3.0

If FSR upscaler could be used as an input by the mod, it mean I would be able to set FSR-FG & upscaler in game but have the mod swap FSR upscaler (only) for the DLSS one.

Looking forward to this!
